quodlibet/quodlibet/formats/_audio.py

```diff
@@ -536,24 +537,77 @@ def _role_call(self, role_tag, sub_keys=None):
 
     @property
     def lyric_filename(self):
-        """Returns the (potential) lyrics filename for this file"""
-
-        filename = self.comma("title").replace(u'/', u'')[:128] + u'.lyric'
-        sub_dir = ((self.comma("lyricist") or self.comma("artist"))
-                  .replace(u'/', u'')[:128])
-
-        if os.name == "nt":
-            # this was added at a later point. only use escape_filename here
-            # to keep the linux case the same as before
-            filename = escape_filename(filename)
-            sub_dir = escape_filename(sub_dir)
-        else:
-            filename = fsnative(filename)
-            sub_dir = fsnative(sub_dir)
-
-        path = os.path.join(
-            expanduser(fsnative(u"~/.lyrics")), sub_dir, filename)
-        return path
+        """Returns the validated, or default, lyrics filename for this
+        file. User defined '[memory] lyric_rootpaths' and
+        '[memory] lyric_filenames' matches take precedent"""
+
+        from quodlibet.pattern import ArbitraryExtensionFileFromPattern
+
+        lyric_paths = \
+            config.getstringlist("memory", "lyric_rootpaths", [])
+        # add default
+        lyric_paths.append(u'~/.lyrics')
+        lyric_filenames = \
+            config.getstringlist("memory", "lyric_filenames", [])
+        # add defaults
+        lyric_filenames.append(
+            (self.comma("lyricist") or self.comma("artist"))
+            .replace(u'/', u'')[:128] + os.sep +
+            self.comma("title").replace(u'/', u'')[:128] + u'.lyric')
+        lyric_filenames.append(
+            (self.comma("lyricist") or self.comma("artist"))
+            .replace(u'/', u'')[:128] + ' - ' +
+            self.comma("title").replace(u'/', u'')[:128] + u'.lyric')
+
+        # generate all paths
+        pathfiles = OrderedDict()
+        pathfiles_expanded = OrderedDict()
+        for p in lyric_paths:
+            for f in lyric_filenames:
+                pathfile = ""
+                if os.name == "nt":
+                    # this was added at a later point. only use
+                    # escape_filename here to keep the linux case
+                    # the same as before
+                    pathfile = \
+                        os.path.join(p, os.path.dirname(f),
+                                     escape_filename(os.path.basename(f)))
+                else:
+                    pathfile = \
+                        os.path.join(p, os.path.dirname(f),
+                                    fsnative(os.path.basename(f)))
+                if not pathfile in pathfiles:
+                    pathfiles[pathfile] = pathfile
+
+        #print_d("looking for lyrics files:\n%s" % '\n'.join(pathfiles.keys()))
+
+        match_ = ""
+        for pathfile in pathfiles.keys():
+            path = expanduser(pathfile)
+            if '<' in path:
```

Comment:
Hmmm, you _can_ create a path with `>` on most OSes, however unlikely it might be...

Reply:
done, note that this 'issue' would only result in extra calls to the builtin 'ArbitraryExtensionFileFromPattern' function ..saves on cycles their, but we've added cycles by switching to regex.

point of me mentioning this detail, is that this code has already slipped into a previous merge #2560  | a64e2d67cad4e65b26f5291fd0006a0942092e4e ..happy to create another to PR to correct this if you'd like??